syntax = "proto3";

package monitor;

service MonitorService {
rpc PingPong (PingRequest) returns (PongResponse) {}
rpc GetMetrics (MetricsRequest) returns (MetricsResponse) {}
rpc Register (RegisterRequest) returns (RegisterResponse) {}
rpc Deregister (DeregisterRequest) returns (DeregisterResponse) {}
}

message PingRequest {
string message = 1;
}

message PongResponse {
string message = 1;
}

message MetricsRequest {
string instance_id = 1;
}

message MetricsResponse {
int32 load = 1;
}

message RegisterRequest {
string instance_id = 1;
}

message RegisterResponse {
bool success = 1;
}

message DeregisterRequest {
string instance_id = 1;
}

message DeregisterResponse {
bool success = 1;
}